pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test: TestJobsProtectedTimestamp failed

pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test.TestJobsProtectedTimestamp [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6094627?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6094627?buildTab=artifacts#/) on master @ [41db784cb97d2749b162020c2c821979094f87b1](https://github.com/cockroachdb/cockroach/commits/41db784cb97d2749b162020c2c821979094f87b1):


```
=== RUN   TestJobsProtectedTimestamp
--- FAIL: TestJobsProtectedTimestamp (1.85s)
=== RUN   TestJobsProtectedTimestamp/system-tenant
    jobs_protected_ts_test.go:178: 
        	Error Trace:	jobs_protected_ts_test.go:129
        	            				jobs_protected_ts_test.go:178
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestJobsProtectedTimestamp/system-tenant
    --- FAIL: TestJobsProtectedTimestamp/system-tenant (0.20s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81185 pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test: TestJobsProtectedTimestamp failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJobsProtectedTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18581

pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test.TestJobsProtectedTimestamp [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6108146?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6108146?buildTab=artifacts#/) on master @ [f4042d47fa8062a612c38d4696eb6bee9cee7c21](https://github.com/cockroachdb/cockroach/commits/f4042d47fa8062a612c38d4696eb6bee9cee7c21):


```
=== RUN   TestJobsProtectedTimestamp
--- FAIL: TestJobsProtectedTimestamp (1.79s)
=== RUN   TestJobsProtectedTimestamp/system-tenant
    jobs_protected_ts_test.go:178: 
        	Error Trace:	jobs_protected_ts_test.go:129
        	            				jobs_protected_ts_test.go:178
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestJobsProtectedTimestamp/system-tenant
    --- FAIL: TestJobsProtectedTimestamp/system-tenant (0.32s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #81185 pkg/ccl/jobsccl/jobsprotectedtsccl/jobsprotectedtsccl_test: TestJobsProtectedTimestamp failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJobsProtectedTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/jobsccl/jobsprotectedtsccl/jobs_protected_ts_test.go
+++ b/pkg/ccl/jobsccl/jobsprotectedtsccl/jobs_protected_ts_test.go
@@ -117,16 +117,14 @@ func testJobsProtectedTimestamp(
 
 	// Verify that the two jobs we just observed as removed were recorded in the
 	// metrics.
-	var removed int
-	runner.QueryRow(t, `
+	runner.CheckQueryResultsRetry(t, `
 SELECT
-    value
+    value >= 2 -- we expect 2, but with retries it can be higher
 FROM
     crdb_internal.node_metrics
 WHERE
     name = 'kv.protectedts.reconciliation.records_removed';
-`).Scan(&removed)
-	require.Equal(t, 2, removed)
+`, [][]string{{"true"}})
 }
 
 // TestJobsProtectedTimestamp is an end-to-end test of protected timestamp

